tests/DataObject/DataObjectDocumentTest.php

```diff
@@ -85,6 +89,40 @@ public function testShouldIndex()
         $this->assertFalse($doc->shouldIndex());
     }
 
+    public function testSubsiteShouldIndex()
+    {
+        $config = $this->mockConfig();
+
+        $config->set(
+            'indexes',
+            [
+                'index0' => [
+                    'subsite_id' => 0,
+                    'includeClasses' => [
+                        VersionedDataObjectFake::class => true
+                    ]
+                ],
+                'index1' => [
+                    'subsite_id' => 1,
+                    'includeClasses' => [
+                        DataObjectFake::class => true
+                    ]
+                ],
+            ]
+        );
+
+        /** @var Versioned $dataobject */
+        $dataobject = new VersionedDataObjectFake([
+            'ID' => 5,
+            'ShowInSearch' => true,
+            'SubsiteID' => null,
+        ]);
+        $dataobject->publishSingle();
+        $doc = DataObjectDocument::create($dataobject);
+
```

Comment:
Can you add a couple more assertions here just to cover more scenarios?

- DataObject with correct SubsiteID should index
- DataObject with incorrect SubsiteID should not index